sched/semaphore/spinlock.c

```diff
@@ -446,4 +446,225 @@ void spin_clrbit(FAR volatile cpu_set_t *set, unsigned int cpu,
 }
 #endif
 
+#ifdef CONFIG_RW_SPINLOCK
+
+/****************************************************************************
+ * Name: read_lock
+ *
+ * Description:
+ *   If this task does not already hold the spinlock, then loop until the
+ *   spinlock is successfully locked.
+ *
+ *   This implementation is non-reentrant and set a bit of lock.
+ *
+ *  The priority of reader is higher than writter if a reader hold the
+ *  lock, a new reader can get its lock but writer can't get this lock.
+ *
+ * Input Parameters:
+ *   lock - A reference to the spinlock object to lock.
+ *
+ * Returned Value:
+ *   None.  When the function returns, the spinlock was successfully locked
+ *   by this CPU.
+ *
+ * Assumptions:
+ *   Not running at the interrupt level.
+ *
+ ****************************************************************************/
+
+void read_lock(FAR volatile rwlock_t *lock)
+{
+  rwlock_t old;
```

Comment:
merge to line 481

Reply:
Is there good reason to make this decision?

Reply:
since it makes the code more compact, and still confirm the c89 spec.

Reply:
I remember C89 need to split declare and using.
It must be declare before using.

Reply:
no, c89 allow declare and assign at the same line, but forbid the declaration in the middle of statements.

Reply:
Ok. I need to some time to rewrite in order to split this PR into two commits.